packages/runtime/src/modules/AttributeListenerModule.ts

```diff
@@ -19,20 +21,68 @@ export class AttributeListenerModule extends RuntimeModule {
         const createdAttribute = event.data;
         if (createdAttribute.content["@type"] === "IdentityAttribute" && createdAttribute.shareInfo) return;
         if (createdAttribute.content["@type"] === "RelationshipAttribute" && createdAttribute.content.confidentiality === RelationshipAttributeConfidentiality.Private) return;
+        if (await AttributeListenerModule.relationshipAttributeOfInactiveRelationship(services, createdAttribute)) return;
 
         const getAttributeListenersResult = await services.consumptionServices.attributeListeners.getAttributeListeners();
         if (getAttributeListenersResult.isError) {
-            this.logger.error("Could not get attribute listeners", getAttributeListenersResult.error);
+            this.logger.error("Could not get LocalAttributeListeners", getAttributeListenersResult.error);
             return;
         }
 
         const attributeListeners = getAttributeListenersResult.value;
         const promises = attributeListeners.map((attributeListener) =>
-            this.createRequestIfAttributeMatchesQuery(services, attributeListener, event.data, event.eventTargetAddress)
+            this.createRequestIfAttributeMatchesQuery(services, attributeListener, createdAttribute, event.eventTargetAddress)
         );
         await Promise.all(promises);
     }
 
+    private static async relationshipAttributeOfInactiveRelationship(services: RuntimeServices, attribute: LocalAttributeDTO): Promise<boolean> {
+        if (attribute.content["@type"] !== "RelationshipAttribute") {
+            return false;
+        }
+
+        const activeRelationshipToPeer = (
+            await services.transportServices.relationships.getRelationships({ query: { peer: attribute.shareInfo?.peer, status: RelationshipStatus.Active } })
+        ).value;
+
+        if (activeRelationshipToPeer.length === 0) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private async handleRelationshipChanged(event: RelationshipChangedEvent) {
+        const changedRelationship = event.data;
+
+        const lastAuditLogEntry = changedRelationship.auditLog.at(-1);
+        if (lastAuditLogEntry && lastAuditLogEntry.reason === RelationshipAuditLogEntryReason.AcceptanceOfCreation) {
```

Comment:
there is always at minimum one entry

Reply:
Do you mean the check could be simplified to `if (lastAuditLogEntry.reason === RelationshipAuditLogEntryReason.AcceptanceOfCreation)`? When removing the check for whether `lastAuditLogEntry` is defined, I receive `'lastAuditLogEntry' is possible 'undefined'.`, even though it is always defined in principle, of course.

Reply:
you can use ! in this case

Reply:
(sorry, I didn't saw this message)